pandas/core/_numba/executor.py

```diff
@@ -15,6 +15,45 @@
 from pandas.compat._optional import import_optional_dependency
 
 
+@functools.cache
+def generate_apply_looper(func, nopython=True, nogil=True, parallel=False):
+    if TYPE_CHECKING:
+        import numba
+    else:
+        numba = import_optional_dependency("numba")
+    nb_compat_func = numba.extending.register_jitable(func)
+
+    @numba.jit(nopython=nopython, nogil=nogil, parallel=parallel)
+    def nb_looper(values, axis):
+        # Operate on the first row/col in order to get
+        # the output shape
+        if axis == 0:
+            first_elem = values[:, 0]
+            dim0 = values.shape[1]
+        else:
+            first_elem = values[0]
+            dim0 = values.shape[0]
+        res0 = nb_compat_func(first_elem)
```

Comment:
Is inferring the shape from the first element similar to what we do for `DataFrame.apply`?

It would be good to note what type of UDFs are supported in the engine docstring

Reply:
``np.apply_along_axis``, which we use does this.

see https://github.com/numpy/numpy/blob/d676a1fe2d495f9d8a86103644bed141c2e69787/numpy/lib/_shape_base_impl.py#L373-L380.

> It would be good to note what type of UDFs are supported in the engine docstring

I'll add a note linking to numba's supported Python/numpy features.